src/proxy.js

```diff
@@ -0,0 +1,278 @@
+
+        return {
+            sessionId,
+            url,
+            groups,
+            country,
+            password,
+            hostname,
+            port,
+        };
+    }
+
+    /**
+     * Returns proxy url.
+     * @param {string} sessionId
+     * Apify Proxy session identifier to be used with requests.
+     * All HTTP requests going through the proxy with the same session identifier
+     * will use the same target proxy server (i.e. the same IP address).
+     * The identifier can only contain the following characters: `0-9`, `a-z`, `A-Z`, `"."`, `"_"` and `"~"`.
+     * @return {string} - the proxy url
+     */
+    getUrl(sessionId) {
+        const username = this._getUsername(sessionId);
+        const { password, hostname, port } = this;
+
+        return `${PROTOCOL}://${username}:${password}@${hostname}:${port}`;
+    }
+
+    /**
+     * Returns proxy username.
+     * @return {string} - the proxy username
+     * @param {string} sessionId
+     * @private
+     */
+    _getUsername(sessionId) {
+        let username;
+        const { groups, country } = this;
+        if (groups || sessionId || country) {
+            const parts = [];
+
+            if (groups && groups.length) {
+                parts.push(`groups-${groups.join('+')}`);
+            }
+            if (sessionId) {
+                parts.push(`session-${sessionId}`);
+            }
+            if (country) {
+                parts.push(`country-${country}`);
+            }
+
+            username = parts.join(',');
+        } else {
+            username = 'auto';
+        }
+        return username;
+    }
+
+    /**
+     * Checks if apify token is provided in env
+     * and gets the password via API and sets it to env
+     * @returns {Promise<void>}
+     * @private
+     */
+    async _setPasswordIfToken() {
+        const token = process.env[ENV_VARS.TOKEN] || LOCAL_ENV_VARS[ENV_VARS.TOKEN];
+        if (!this.password) {
+            if (token) {
+                const { proxy: { password, groups } } = await apifyClient.users.getUser({ token, userId: 'me' });
+                this.password = password;
+                this.availableGroups = groups;
+            } else {
+                throw new Error(this._getMissingParamErrorMgs('password', ENV_VARS.PROXY_PASSWORD));
+            }
+        }
+    }
+
+    /**
+     * Checks the status of apify proxy and
+     * throws an access denied error if the status is not "connected".
+     * Also checks if all proxy groups are available if token is provided.
+     * @returns {Promise<void>}
+     * @private
+     */
+    async _checkAccess() {
+        // Check if user has access to selected proxy groups
+        await this._checkAccessToProxyGroups();
+
+        const url = APIFY_PROXY_STATUS_URL;
+        const proxyUrl = this.getUrl();
+        const countryCode = this.country;
+        const { body } = await requestAsBrowser({ url, proxyUrl, countryCode });
```

Comment:
you can add `json: true` here to get parsed JSON in `body`

Reply:
Yop... removed the parsing part.